partitioned-heat-conduction/fenics/heat.py

```diff
@@ -160,14 +159,14 @@ def determine_gradient(V_g, u, flux):
 bc = []
 # Create for each dimension of Vbig a coupling expression for either the time derivatives (Dirichlet side)
 #  or Neumann side (no time derivatives required as they are enforced with changing F)
-coupling_expressions = [precice.create_coupling_expression()] * tsm.num_stages
+coupling_expressions = [precice.create_coupling_expression() for _ in range(tsm.num_stages)]
```

Comment:
This was definitely a nasty bug: `[something] * N` creates a list with `N` references to the same object. You can check this by printing `coupling_expressions` and looking at the names of the expressions.

Still: I do not get the correct solution, but this should be a step into the right direction.

Reply:
This bug was actually hiding another bug with the Neumann BCs. By fixing this bug the Neumann BC bug became visible. This also caused the (previously working) Fixme 4 part to fail. See below for comment on the Neumann BC bugfix.